crates/precompile/src/secp256k1.rs

```diff
@@ -84,8 +31,16 @@ pub fn ec_recover_run(input: &Bytes, gas_limit: u64) -> PrecompileResult {
     let recid = input[63] - 27;
     let sig = <&B512>::try_from(&input[64..128]).unwrap();
 
-    let out = secp256k1::ecrecover(sig, recid, msg)
-        .map(|o| o.to_vec().into())
-        .unwrap_or_default();
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "secp256k1")] {
+            let res = bitcoin_secp256k1::ecrecover(sig, recid, msg);
+        } else if #[cfg(feature = "libsecp256k1")] {
+            let res = parity_libsecp256k1::ecrecover(sig, recid, msg);
+        } else {
+            let res = k256::ecrecover(sig, recid, msg);
+        }
```

Comment:
This is were switch is happening, priority is secp256k1 (bitcoin lib), parity (if enabled) and by default k256.